tests/test_formatting.py

```diff
@@ -402,3 +402,17 @@ def test_issue_104(self):
             ]
         })
         self.assertEqual(result, expected)
+
+    def test_with_cte(self):
+        expected = "WITH t AS (SELECT a FROM table) SELECT * FROM t"
+        result = format({'select': '*', 'from': 't', 'with': {'name': 
+                         't', 'value': {'select': {'value': 'a'}, 'from': 'table'}}
+                        })
+        self.assertEqual(result, expected)
+
+    def test_with_cte_various(self):
+        expected = "WITH t1 AS (SELECT a FROM table), t2 AS (SELECT 1) SELECT * FROM t1, t2"
+        result = format({'select': '*', 'from': ['t1', 't2'], 
+                         'with': [{'name': 't1', 'value': {'select': {'value': 'a'}, 'from': 'table'}}, 
+                                  {'name': 't2', 'value': {'select': {'value': 1}}}]})
+        self.assertEqual(result, expected)
```

Comment:
Excellent.  I wanted to see more-than-one definition

Reply:
Thank you for the review ;)